### ISSUE TITLE
Bug with JsonSchema::nullOr combined with JsonSchema::enum

### ISSUE THREAD

**Issue by martin-schilling** [event-engine/php-json-schema#4]
Today I noticed a bug where it seemed like `JsonSchema::nullOr` would not take effect when used on `JsonSchema::enum`. I would expect a null value to be accepted in this case, but it was rejected instead. I will probably look into that and provide a fix.

**Comment by codeliner** [comment]
yeah, sounds like a bug. Looking forward to a fix :+1:

**Comment by martin-schilling** [comment]
Ok it seems like I found the issue. In the current version the following code would produce as schema like this:
```
$schema = \EventEngine\JsonSchema\JsonSchema::nullOr(
    \EventEngine\JsonSchema\JsonSchema::enum(['ab', 'cd'])
);
var_dump($schema->toArray());
```
```
array(2) {
  ["type"]=>
  array(2) {
    [0]=>
    string(6) "string"
    [1]=>
    string(4) "null"
  }
  ["enum"]=>
  array(2) {
    [0]=>
    string(2) "ab"
    [1]=>
    string(2) "cd"
  }
}
```
A validation with a null value against this schema would produce the following error:
```
[enum] Does not have a value in the enumeration ["ab","cd"]
```
I fixed this issue by altering the created schema to be the following:
```
array(2) {
  ["type"]=>
  array(2) {
    [0]=>
    string(6) "string"
    [1]=>
    string(4) "null"
  }
  ["enum"]=>
  array(3) {
    [0]=>
    string(2) "ab"
    [1]=>
    string(2) "cd"
    [2]=>
    NULL
  }
}
```

Please look at the changes properly since I do not have a lot of experience with JsonSchema.
https://github.com/event-engine/php-json-schema/pull/5

**Comment by martin-schilling** [comment]
Some references to other issues including different solutions for this problem. I followed the first one:
https://github.com/json-schema-org/json-schema-spec/issues/258
https://github.com/epoberezkin/ajv/issues/824

**Comment by codeliner** [comment]
your solution looks good. I'm not an expert here either, but the solution is discussed in details in the linked issue so I'd say it is correct. Merged your PR #5 already
